x-pack/plugins/security/common/model/user_profile.ts

```diff
@@ -65,7 +65,7 @@ export interface UserProfileUserInfo {
 /**
  * Placeholder for data stored in user profile.
  */
-export type UserProfileData = Record<string, unknown>;
+export type UserProfileData = Record<any, any>;
```

Comment:
This change was needed to allow more specific TS than a generic object.
See here: https://github.com/elastic/kibana/pull/161214/files#diff-f8e7bc19afaf6d981dd4971618f887d5b4600fa1e9dd0035859363ec3ca1e049R67

Reply:
We should avoid usage of `any` wherever possible. Can you explain what errors you got when using `Record<string, unknown>`?

Reply:
The fact that the Object key must be `string` we can't have a more specific Type to work with.

This is the type I want to work with

```ts
interface UserProfileData {
  avatar?: UserProfileAvatarData;
  userSettings?: UserSettingsData;
}
```

But that type can't extend `Record<string, unknown>`

I wanted to be more specific on this file (`x-pack/plugins/security/common/model/user_profile.ts`) but you are using a different interface for client and the server. On the server you add an extra `kibana` key to the object. (I think this should be refactored and better typed as the client and the server don't have the same interface. But that was out of scope for this PR).

Reply:
EDIT: I reverted the change and extended my interface: https://github.com/elastic/kibana/pull/161214/commits/0ca82e172b21beb3feeae75641aa2a3d10e798bb

Reply:
Got you. 

I'm not sure why TS is ok with us passing the same interface inline but throws an error when you define it as an interface but that was not the intention and consumers should be able to do both:

```ts
type UserProfileData = Record<string, unknown>;

interface SpecificUserProfileData {
  avatar?: {};
  userSettings?: {};
}

function updateData<T extends UserProfileData>(data: T) {}

// inline passes
updateData<{
  avatar?: {};
  userSettings?: {};
}>({})

// interface fails
updateData<SpecificUserProfileData>({})
```

The purpose of the type is simply to enforce an object so happy to go with your approach or removing it altogether and changing the generics to extend from `{}`.

Reply:
For now adding the key declaration in my interface works.